R/stash.R

```diff
@@ -48,26 +50,31 @@ stash <- function(var, code, depends_on = NULL) {
   if (has_been_stashed(var)) {
     old_hash_tbl <- get_hash_table(var)
     if (hash_tables_are_equivalent(old_hash_tbl, new_hash_tbl)) {
-      message("Loading stashed object.")
-      load_variable(var)
+      if(verbose) message("Loading stashed object.")
+      res <- load_variable(var, functional)
     } else {
-      message("Updating stash.")
-      new_stash(var, formatted_code, new_hash_tbl)
+      if(verbose) message("Updating stash.")
+      res <- new_stash(var, formatted_code, new_hash_tbl, functional)
     }
   } else {
-    message("Stashing object.")
-    new_stash(var, formatted_code, new_hash_tbl)
+    if(verbose) message("Stashing object.")
+    res <- new_stash(var, formatted_code, new_hash_tbl, functional)
   }
-
-  invisible(NULL)
+  
+  invisible(res)
 }
 
 # Make a new stash from a variable, code, and hash table.
-new_stash <- function(var, code, hash_tbl) {
+new_stash <- function(var, code, hash_tbl, functional) {
   val <- evaluate_code(code)
-  assign_value(var, val)
   write_hash_table(var, hash_tbl)
   write_val(var, val)
+  if(functional) {
```

Comment:
Can you add explicit return calls? I think that would make the code a bit clearer and could potentially prevent some unexpected behavior (especially if new features are added).